wgsl/index.bs

```diff
@@ -1697,6 +1700,327 @@ Issue: (dneto): Do we have to explicitly list the type environment Gamma? That's
+  <tr><td>*e1* : *T*<br> *e2* : *T*<br> *T* is *IntVec* with unsigned component<td>`e1 % e2` : *T*<td>Component-wise unsigned integer modulus (OpUMod)
+  <tr><td>*e1* : *T*<br> *e2* : *T*<br> *T* is *IntVec* with signed component<td>`e1 % e2` : *T*<td>Component-wise signed integer remainder (OpSMod)
+  <tr><td>*e1* : *T*<br> *e2* : *T*<br> *T* is *FloatVec*<td>`e1 % e2` : *T*<td>Component-wise floating point division (OpFMod)
+</table>
+
+<table class='data'>
+  <caption>Binary arithmetic expressions with mixed scalar, vector, and matrix operands</caption>
+  <thead>
+    <tr><td>Precondition<td>Conclusion<td>Notes
+  </thead>
+  <tr><td>*e1* : f32<br>
+          *e2* : *T*<br>
+          *T* is *FloatVec*
+      <td>`e1 * e2` : *T*<br>
+          `e2 * e1` : *T*
+      <td>Scalar multiplication of vector (OpVectorTimesScalar)
+  <tr><td>*e1* : f32<br>
+          *e2* : *T*<br>
+          *T* is mat*N*x*M*&lt;f32&gt;
+      <td>`e1 * e2` : *T*<br>
+          `e2 * e1` : *T*
+      <td>Scalar multiplication of matrix (OpMatrixTimesScalar)
+  <tr><td>*e1* : vec*M*&lt;f32&gt;<br>
+          *e2* : mat*N*x*M*&lt;f32&gt;
+      <td>`e1 * e2` : vec*N*&lt;f32&gt;<br>
+      <td>Vector times matrix (OpVectorTimesMatrix)
+  <tr><td>*e1* : mat*N*x*M*&lt;f32&gt;<br>
+          *e2* : vec*N*&lt;f32&gt;
+      <td>`e1 * e2` : vec*M*&lt;f32&gt;<br>
+      <td>Matrix times vector (OpMatrixTimesVector)
+  <tr><td>*e1* : mat*K*x*N*&lt;f32&gt;<br>
+          *e2* : mat*M*x*K*&lt;f32&gt;<br>
+      <td>`e1 * e2` : mat*M*x*N*&lt;f32&gt;<br>
+      <td>Matrix times matrix (OpMatrixTimesMatrix)
+</table>
+
+<table class='data'>
+  <caption>Bit shift expressions</caption>
+  <thead>
+    <tr><td>Precondition<td>Conclusion<td>Notes
+  </thead>
+  <tr><td>*e1* : *T*<br>
+          *e2* : *T*<br>
+          *T* is *Int*
+       <td class="nowrap">`e1 << e2` : *T*
+       <td>Shift *e1* left by *e2* bits (OpShiftLeftLogical)
+  <tr><td>*e1* : *T*<br>
+          *e2* : *T*<br>
+          *T* is *IntVec*
+       <td class="nowrap">`e1 << e2` : *T*
+       <td>Component-wise shift left (OpShiftLeftLogical)
+  <tr><td>*e1* : u32<br>
+          *e2* : u32<br>
+       <td class="nowrap">`e1 >> e2` : u32
+       <td>Logical shift *e1* right by *e2* bits, i.e. inserting zero at most significant bits (OpShiftRightLogical)
+  <tr><td>*e1* : *T*<br>
+          *e2* : *T*<br>
+          *T* is vec*N*&lt;u32&gt;
+       <td class="nowrap">`e1 >> e2` : *T*
+       <td>Component-wise logical shift right (OpShiftRightLogical)
+  <tr><td>*e1* : i32<br>
+          *e2* : i32<br>
+       <td class="nowrap">`e1 >> e2` : u32
+       <td>Arithmetic shift *e1* right by *e2* bits, i.e. replicating the sign bit of *e1* at most significant bits (OpShiftRightArithmetic)
+  <tr><td>*e1* : *T*<br>
+          *e2* : *T*<br>
+          *T* is vec*N*&lt;i32&gt;
+       <td class="nowrap">`e1 >> e2` : *T*
```

Comment:
ditto

Reply:
Sorry, I've lost the context now.

Reply:
Usually it's a ditto to the comment above, so I'm guessing this is in reference to >> vs >>>?